File: src/events/messageCreate.js
import { handleCommand } from '../commands/index.js';
import logger from '../utils/logger.js';

export const name = 'messageCreate';

export async function execute(message) {
  // Ignore bot messages
  if (message.author.bot) return;

  // Try to handle as a command
  try {
    await handleCommand(message);
  } catch (err) {
    logger.error('Error handling command', { error: err.message });
  }
}

export default { name, execute };

File: src/events/message_create.py
"""Message create event handler."""
import discord
from src.commands.registry import handle_command
from src.utils.logger import logger


async def execute(message: discord.Message, client: discord.Client):
    """Execute message create event."""
    if message.author.bot:
        return
    try:
        await handle_command(message, client)
    except Exception as err:
        logger.error('Error handling command', {'error': str(err)})
